layers.go

```diff
@@ -857,6 +878,7 @@ func (r *layerStore) Delete(id string) error {
 		if layer.MountPoint != "" {
 			delete(r.bymount, layer.MountPoint)
 		}
+		r.deleteInDigestMap(id)
```

Comment:
Probably outside of the scope of this PR, but is there any advantage to backport the deleteInternal() function from master where this particular function is called from there?

Reply:
> Probably outside of the scope of this PR, but is there any advantage to backport the deleteInternal() function from master where this particular function is called from there?

Good point. Tried to look into it...
AFAICS the `deleteInternal() `just contains the core deletion operations once in `Delete()`.
The main difference  I can see in commit [bcedb54d05](https://github.com/containers/storage/commit/bcedb54d0547a9a5c0c641518a86a72efc2b19dd) on master (which introduced the function) is that now the `Load()` function calls `deleteInternal()`  in place of the full `Delete()` function.
This skips from the `Load()` the operations left in `Delete()`, which are:
* the checks if the layers are mounted
* the call to the `Save()` function

As [bcedb54d05](https://github.com/containers/storage/commit/bcedb54d0547a9a5c0c641518a86a72efc2b19dd) commit message says, this was introduced to skip the mountpoints locking in the `Save()` function, as locking already happens in `Load()`.
Looking at the code in the 1.11 branch, the locking management seems quite different. In particular the `Save()` function doesn't perform any lock.
So, my guess is that we wouldn't benefit from the `deleteInternal()` function here (moreover, backporting wouldn't be trivial).

Reply:
Thanks for taking a look @fgiudici.  I was afraid that once you undid one string, too much would unravel with that.